scripts/ppr.py

```diff
@@ -0,0 +1,131 @@
+import psycopg2
+import requests
+import json
+import os
+
+def insertPath(cur, relation_id, dhid_from, dhid_to, path):
+    stepList = [f"{step[0]} {step[1]}" for step in path]
+    linestring = "LINESTRING(" + ",".join(stepList) + ")"
+    cur.execute(
+        'INSERT INTO paths (stop_area_relation_id, "from", "to", geom) VALUES (%s, %s, %s, ST_GeomFromText(%s, 4326))',
+        (relation_id, dhid_from, dhid_to, linestring)
+    )
+
+
+def insertPathsElementsRef(cur, edges, path_counter):
+    osm_way_ids = []
+    for edge in edges:
+        osm_way_id = abs(edge["osm_way_id"])
+        # currently also the crossed ways are included (negative way osm id)
+        if osm_way_id != 0 and osm_way_id not in osm_way_ids:
+            osm_way_ids.append(osm_way_id)
+            cur.execute(
+                "INSERT INTO paths_elements_ref (path_id, osm_type, osm_id) VALUES (%s, %s, %s)",
+                (path_counter, 'W', osm_way_id)
```

Comment:
PPR is not explicitly returning the osm type. For example an edge using an elevator is returned as following:
```
{
    "edge_type": "elevator",
    "street_type": "none",
    "crossing_type": "generated",
    "side": "center",
    "osm_way_id": -9121043127,
    "name": "",
    "marked_crossing_detour": 0,
    "distance": 0.0,
    ...
},
```

These are the edge types that PPR uses:

>     CONNECTION
>     STREET
>     FOOTWAY
>     CROSSING
>     ELEVATOR

Crossings always return the way that is crossed, even if there is a crossing node, so I think that elevator will be the only type that uses nodes. It may be possible to distinguish between ways and nodes in this way. But this also means, that the tags from crossing nodes get lost (information regarding traffic_signals, ...).